### PR TITLE
allow to set attributes on bold and italic

### PR DESCRIPTION
**Pull request by gabrielhase**
@peyerluk This is just a rough proposal. It was quicker doing it in code 😄 
Give me a go if you're happy with the concept and I'll implement this for real.

### CODE REVIEW

File: src/cursor.js [hunk 1]
```diff
@@ -237,8 +237,12 @@ export default class Cursor {
 
   // Create an element with the correct ownerWindow
   // (see: http://www.w3.org/DOM/faq.html#ownerdoc)
-  createElement (tagName) {
-    return this.win.document.createElement(tagName)
+  createElement (tagName, attributes = []) {
+    const element = this.win.document.createElement(tagName)
+    for (const {attributeName, attributeValue} of attributes) {
```
Comment (marcbachmann) [inline src/cursor.js]: I think that an object for the attributes is more common.
It's closer to the dom api. I'd prefer that. `attributes: {class: 'foo'}`

**Comment by gabrielhase** [comment]
@peyerluk I need your feedback here to proceed.

File: src/config.js [hunk 2]
```diff
@@ -9,7 +9,15 @@ export const logErrors = true
 export const editableClass = 'js-editable'
 export const editableDisabledClass = 'js-editable-disabled'
 export const pastingAttribute = 'data-editable-is-pasting'
-export const boldTag = 'strong'
+export const boldMarkup = {
```
Comment (peyerluk) [inline src/config.js]: We can use the same syntax as in the framework:
```js
export const boldMarkup = {
  type 'tag'
  name: 'strong'
  attribs: {
    'class': 'foo'
  }
}
```

File: src/cursor.js (hunk 1 above)
Comment (peyerluk) [inline src/cursor.js]: 👍 and see my comment above in the config.js

File: src/config.js (hunk 2 above)
Comment (gabrielhase) [inline src/config.js]: 👍  will do

File: src/cursor.js [hunk 3]
```diff
@@ -239,7 +239,9 @@ export default class Cursor {
   // (see: http://www.w3.org/DOM/faq.html#ownerdoc)
   createElement (tagName, attributes = []) {
     const element = this.win.document.createElement(tagName)
-    for (const {attributeName, attributeValue} of attributes) {
+    // I'd like to use for .. of with Object.entries() here but no IE support...
+    for (const attributeName of Object.keys(attributes)) {
```
Comment (marcbachmann) [inline src/cursor.js]: Can't you just use `for key in object`?
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/for...in

```
for (const key in attributes) element.setAttribute(key, attributes[key])
```

Comment (gabrielhase) [inline src/cursor.js]: I could, but do you really think that is nicer?

Comment (marcbachmann) [inline src/cursor.js]: definitely. you can keep the braces and newlines if you like 😉

Comment (marcbachmann) [inline src/cursor.js]: And please fix the fallback
```
createElement (tagName, attributes = {}) {...
```

Comment (gabrielhase) [inline src/cursor.js]: good catch, thx!

**Comment by gabrielhase** [comment]
@peyerluk please check again.

File: src/config.js [hunk 4]
```diff
@@ -9,8 +9,27 @@ export const logErrors = true
 export const editableClass = 'js-editable'
 export const editableDisabledClass = 'js-editable-disabled'
 export const pastingAttribute = 'data-editable-is-pasting'
-export const boldTag = 'strong'
-export const italicTag = 'em'
+export const boldMarkup = {
+  type: 'tag',
+  name: 'strong',
+  attribs: {
+    'class': 'foo'
```
Comment (peyerluk) [inline src/config.js]: Please remove

File: src/config.js [hunk 5]
```diff
@@ -9,8 +9,27 @@ export const logErrors = true
...
+  }
+}
+export const italicMarkup = {
+  type: 'tag',
+  name: 'em',
+  attribs: {
+    'class': 'bar'
```
Comment (peyerluk) [inline src/config.js]: same here

File: src/config.js [hunk 6]
```diff
@@ -9,8 +9,27 @@ export const logErrors = true
...
+  }
+}
+export const linkMarkup = {
+  type: 'tag',
+  name: 'a',
+  attribs: {
+    'class': 'foo bar'
```
Comment (peyerluk) [inline src/config.js]: and here

Comment (gabrielhase) [inline src/config.js]: really, this one too? But it's `foo bar` 😆

**Pull request merged**